spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
+			}
+
+			@Override
+			public void handleConsumeOk(String consumerTag) {
+				super.handleConsumeOk(consumerTag);
+				latch.countDown();
+			}
+
+		};
+		channel.basicConsume(queueName, consumer);
+		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
@garyrussell , here is a polishing to your PR: https://github.com/artembilan/spring-amqp/commit/6ff39d359993b7c21cdad3457fed3de6a08df136

I don't see reason to raise a new JIRA for that test, just because we don't have any concern.

If the `handleCancel()` is invoked on the `QueueingConsumer`, its `handle(Delivery delivery)` will take care about that `POISON` and the `com.rabbitmq.client.ConsumerCancelledException` is thrown to the `RabbitTemplate.receive()` thread to be handled as normal in the `execute()`'s `catch` block.

Note: pay attention to the `TODO` in my commit. I'd like to hear your point.